sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ContinuesMessageSubscriber.java

```diff
@@ -0,0 +1,322 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.util.logging.ClientLogger;
+import org.reactivestreams.Subscription;
+import reactor.core.CoreSubscriber;
+import reactor.core.Exceptions;
+import reactor.core.publisher.FluxProcessor;
+import reactor.core.publisher.FluxSink;
+import reactor.core.publisher.Operators;
+import reactor.util.context.Context;
+
+import java.util.Deque;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+
+/**
+ * Processor that listens to upstream messages, pushes them downstream then completes it if necessary.
+ */
+class ContinuesMessageSubscriber extends FluxProcessor<ServiceBusReceivedMessage, ServiceBusReceivedMessage> implements Subscription{
```

Comment:
Is there a reason you wrote a Processor rather than extending from BaseSubscriber or leveraging an existing one from the framework?

Reply:
It probably can be implemented with BaseSubscriber but I have not tried it.

Reply:
moved this code out to separate branch